front/src/Phaser/Game/GameMap.ts

```diff
@@ -100,4 +115,46 @@ export class GameMap {
         }
         callbacksArray.push(callback);
     }
+
+    public findLayer(layerName: string): ITiledMapLayer | undefined {
+        let i = 0;
+        let found = false;
+        while (!found && i<this.flatLayers.length) {
+            if (this.flatLayers[i].name === layerName) {
+               found = true;
+            }
+            else {
+                i++;
+            }
+        }
+        if (found) {
+            return this.flatLayers[i];
+        }
+        return undefined;
+    }
+
+    public findPhaserLayer(layerName: string): TilemapLayer | undefined {
+        let i = 0;
+        let found = false;
+        while (!found && i<this.phaserLayers.length) {
+            if (this.phaserLayers[i].layer.name === layerName) {
+                found = true;
+            }
+            else {
+                i++;
+            }
+        }
+        if (found) {
+            return this.phaserLayers[i];
+        }
+        return undefined;
```

Comment:
Idem, can't we express this as:

```suggestion
    return this.phaserLayers.find((layer) => layer.layer.name = layerName);
```

?